ravif/src/av1encoder.rs

```diff
@@ -216,49 +226,30 @@ impl Encoder {
             ColorModel::YCbCr => MatrixCoefficients::BT601,
             ColorModel::RGB => MatrixCoefficients::Identity,
         };
-        match self.depth {
-            BitDepth::Eight | BitDepth::Auto => {
-                let planes = buffer.pixels().map(|px| {
-                    let (y, u, v) = match self.color_model {
-                        ColorModel::YCbCr => rgb_to_8_bit_ycbcr(px.rgb(), BT601),
-                        ColorModel::RGB => rgb_to_8_bit_gbr(px.rgb()),
-                    };
-                    [y, u, v]
-                });
-                let alpha = buffer.pixels().map(|px| px.a);
-                self.encode_raw_planes_8_bit(width, height, planes, Some(alpha), PixelRange::Full, matrix_coefficients)
-            },
-            BitDepth::Ten => {
-                let planes = buffer.pixels().map(|px| {
-                    let (y, u, v) = match self.color_model {
-                        ColorModel::YCbCr => rgb_to_10_bit_ycbcr(px.rgb(), BT601),
-                        ColorModel::RGB => rgb_to_10_bit_gbr(px.rgb()),
-                    };
-                    [y, u, v]
-                });
-                let alpha = buffer.pixels().map(|px| to_ten(px.a));
-                self.encode_raw_planes_10_bit(width, height, planes, Some(alpha), PixelRange::Full, matrix_coefficients)
-            },
-        }
+        let planes = buffer.pixels().map(|px| match self.color_model {
+            ColorModel::YCbCr => rgb_to_ycbcr(px.rgb(), self.depth, BT601),
+            ColorModel::RGB => [px.g, px.b, px.r],
+        });
+        let alpha = buffer.pixels().map(|px| px.a);
+        self.encode_raw_planes(width, height, planes, Some(alpha), PixelRange::Full, matrix_coefficients)
     }
 
-    fn convert_alpha(&self, in_buffer: Img<&[RGBA8]>) -> Option<ImgVec<RGBA8>> {
+    fn convert_alpha<P: Pixel + Default>(&self, in_buffer: Img<&[Rgba<P>]>) -> Option<ImgVec<Rgba<P>>> {
         match self.alpha_color_mode {
             AlphaColorMode::UnassociatedDirty => None,
             AlphaColorMode::UnassociatedClean => blurred_dirty_alpha(in_buffer),
             AlphaColorMode::Premultiplied => {
-                let prem = in_buffer.pixels()
-                    .filter(|px| px.a != 255)
+                let prem = in_buffer
+                    .pixels()
+                    .filter(|px| px.a != P::cast_from(255))
                     .map(|px| {
-                        if px.a == 0 {
-                            RGBA8::default()
+                        if Into::<u32>::into(px.a) == 0 {
+                            Rgba::new(px.a, px.a, px.a, px.a)
```

Comment:
Why change the default background from black to white?

Reply:
I don't understand how you interpret this as setting the default background from black to white.

Reply:
This is only executed when `px.a == 0` so it should be `[0, 0, 0, 0]` just like what `::default()` used to emit.

It was never `[0, 0, 0, a]`either.